### PR TITLE
Add validation for patient transfers

### PR DESCRIPTION
**Pull request by Ashesh3**
Related Issue: https://github.com/coronasafe/care_fe/issues/7662
Frontend: https://github.com/coronasafe/care_fe/pull/7689

This pull request adds validation for patient transfers, including checking the year of birth and the last consultation discharge date. It also ensures that the last consultation discharge date is after the last encounter date.

### CODE REVIEW

**Review by sainak — APPROVED** [review]

File: care/facility/api/serializers/patient_consultation.py [hunk 1]
```diff
@@ -515,6 +515,22 @@ def validate_encounter_date(self, value):
...
+            .first()
+        )
+        if last_consultation and value < last_consultation.encounter_date:
+            raise ValidationError(
+                {
+                    "encounter_date": [
+                        "This field value must be greater than the last discharge date."
+                    ]
+                }
+            )
```
Comment (sainak) [inline care/facility/api/serializers/patient_consultation.py]: move this to validate() as it depends on multiple fields

Comment (Ashesh3) [reply care/facility/api/serializers/patient_consultation.py]: Done

**Review by sainak — APPROVED** [review]

**Comment by nihal467** [comment]
LGTM

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
marked as do not merge for getting some clarity from aparna

**Comment by vigneshhari** [comment]
Closing PR due to inactivity

**Pull request closed without merging**